azkaban-common/src/main/java/azkaban/container/models/AzKubernetesV1PodBuilder.java

```diff
@@ -0,0 +1,189 @@
+ *
+ * This class provides custom builder methods to have tight coupling of
+ * volumes and init containers (associated with job-types) with application-container
+ * enabling better type safety and compact implementation
+ *
+ * The Pod-spec created using this builder creates one application-container
+ * and multiple init containers corresponding to each jobType
+ */
+public class AzKubernetesV1PodBuilder {
+    private static final String JOBTYPE_VOLUME_PREFIX = "jobtype-volume-";
+    private static final String JOBTYPE_INIT_PREFIX = "jobtype-init-";
+    private static final String AZ_CLUSTER_KEY = "AZ_CLUSTER";
+    private static final String AZ_CONF_VERSION_KEY = "AZ_CONF_VERSION";
+    private static final String JOBTYPE_MOUNT_PATH_KEY = "JOBTYPE_MOUNT_PATH";
+    private static final String DEFAULT_RESTART_POLICY = "Never";
+    private static final String API_VERSION = "v1";
+    private static final String KIND = "Pod";
+
+    private final V1ObjectMetaBuilder podMetaBuilder = new V1ObjectMetaBuilder();
+    private final V1ContainerBuilder flowContainerBuilder = new V1ContainerBuilder();
+
+    private final List<V1VolumeMount> appVolumeMounts = new ArrayList<>();
+    private final List<V1Volume> appVolumes = new ArrayList<>();
+    private final List<V1Container> initContainers = new ArrayList<>();
+
+    private final V1EnvVar azClusterName;
+    private final String restartPolicy;
+
+    /**
+     * @param podName Name of the kubernetes "Pod" resource unique within the namespace
+     * @param podNameSpace Name of the kubernetes "Pod" namespace
+     * @param azClusterName Name to uniquely represent Azkaban instance requesting creation of Pod
+     * @param restartPolicy Optional argument to specify flow container restart policy, otherwise, its "Never"
+     */
+    public AzKubernetesV1PodBuilder(String podName, String podNameSpace, String azClusterName, Optional<String> restartPolicy) {
+        this.podMetaBuilder
+                .withName(podName)
+                .withNamespace(podNameSpace);
+        this.azClusterName = new V1EnvVarBuilder()
+                .withName(AZ_CLUSTER_KEY)
+                .withValue(azClusterName)
+                .build();
+        this.restartPolicy = restartPolicy.orElse(DEFAULT_RESTART_POLICY);
+    }
+
+    /**
+     * @param labels Key-Val containing identifying information and can be utilized
+     *               for the kubernetes selector queries
+     */
+    public AzKubernetesV1PodBuilder withPodLabels(Map<String, String> labels) {
+        this.podMetaBuilder.withLabels(labels);
+        return this;
+    }
+
+    /**
+     * @param annotations Key-Val containing non-identifying information
+     */
+    public AzKubernetesV1PodBuilder withPodAnnotations(Map<String, String> annotations) {
+        this.podMetaBuilder.withAnnotations(annotations);
+        return this;
+    }
+
+    /**
+     * @param name Flow-container/ application-container name
```

Comment:
Can you please add some documentation about what method is doing?

Reply:
Done